cloudify_docker/tasks.py

```diff
@@ -922,22 +918,23 @@ def handle_container_timed_out(ctx, docker_client, container_args,
             "Stop Contianer {0} from tag {1} with command {2}".format(
                 container, image_tag, stop_command))
         # attach to container socket and send the stop_command
-        socket = docker_client.attach_socket(container,
-                                             params={
-                                                 'stdin': 1,
-                                                 "stdout": 1,
-                                                 'stream': 1,
-                                                 "logs": 1
-                                             })
+        container_obj = docker_client.containers.get(container)
+        socket = container_obj.attach_socket(
+            params={
+                'stdin': 1,
+                "stdout": 1,
+                'stream': 1,
+                "logs": 1
+            })
         try:
-            socket.settimeout(20)  # timeout for 20 seconds
-            socket.send(stop_command)
+            socket._sock.settimeout(20)
```

Comment:
Should we expose this to the user?

Reply:
we have no other choice [ besides , that is the only way or the function will hang indefinitely ]